app/src/main/java/bg/zahov/app/ui/settings/SettingsViewModel.kt

```diff
@@ -1,74 +1,101 @@
 package bg.zahov.app.ui.settings
 
-import android.app.Application
-import androidx.lifecycle.AndroidViewModel
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import bg.zahov.app.Inject
 import bg.zahov.app.data.local.Settings
-import bg.zahov.app.getSettingsProvider
-import bg.zahov.app.getUserProvider
-import bg.zahov.app.getWorkoutStateManager
-import bg.zahov.fitness.app.R
+import bg.zahov.app.data.provider.SettingsProviderImpl
+import bg.zahov.app.data.provider.UserProviderImpl
+import bg.zahov.app.data.provider.WorkoutStateManager
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
-class SettingsViewModel(application: Application) : AndroidViewModel(application) {
+/**
+ * ViewModel for managing app settings.
+ *
+ * @property repo Provides access to the user's settings via [SettingsProviderImpl].
+ * @property auth Manages user authentication and provides user actions via [UserProviderImpl].
+ * @property workoutState Manages the current workout state, ensuring any ongoing workout is tracked and can be cancelled on user logout or account deletion.
+ *
+ * This ViewModel handles:
+ * - Retrieving and updating user settings.
+ * - Logging out the user and resetting their settings.
+ * - Deleting the user's account and associated data.
+ */
+class SettingsViewModel(
+    private val repo: SettingsProviderImpl = Inject.settingsProvider,
```

Comment:
Use the interfaces here instead of the impl classes.

Reply:
Changed it